docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,429 @@
+```
+git status
+```
+
+![submit_pr_squash_commits_2.png](./imgs/submit_pr_squash_commits_2.png)
+
+I do have my feature branch checked out, and it is in synch with the remote
+feature branch.
+
+Now let's rebase the feature branch off of the master branch in an interactive
+mode that allows us to `squash` the commits.
+
+Run :
+
+```
+git rebase -i master
+```
+
+![submit_pr_squash_commits_3.png](./imgs/submit_pr_squash_commits_3.png)
+
+The `vim` editor opens and lists my 3 commits, and some commands that are
+available.
+
+Type `i` to place the editor into `INSERT` mode. Use the `arrow` keys to move
+around, and the `delete` or `backspace` key for deleting characters.
+
+I want to keep the commit message from the first commit, so I will leave the
+word `pick` in front of the first commit, and change the words from `pick` to
+`squash` for the second and third commits.
+
+To save these changes, press the `escape` key, and type `:wq`
+
+![submit_pr_squash_commits_4.png](./imgs/submit_pr_squash_commits_4.png)
+
+The `vim` editor comes up again to give me a chance to change my commit message.
+You'll notice that because I previously chose `squash` for the second and third
+commits, it appended the commit messages of the second and third commits onto
+the commit message of the first.
+
+![submit_pr_squash_commits_5.png](./imgs/submit_pr_squash_commits_5.png)
+
+This can give a user a chance to view all messages before formulating one final
+message.
+
+If I had a large number of commits and didn't want all commit messages to be
+appended together, I would have specified: `pick, fixup, fixup` instead of
+`pick, squash, squash` . The command `fixup` is like `squash`, except that it
+discards a commit's log message.
+
+In my case, I only like the message of the first commit, so I will delete the
+others.
+
+![submit_pr_squash_commits_6.png](./imgs/submit_pr_squash_commits_6.png)
+
+Save the changes, and the interactive `rebase` command finally completes.
+
+Push the changes from the local feature branch to the remote feature branch.
+
+Run :
+
+```
+git push --force-with-lease
+```
+
+The remote feature branch now has one commit.
+
+![submit_pr_squash_commits_7.png](./imgs/submit_pr_squash_commits_7.png)
+
+### 8. Create the pull request (PR)
+
+In your browser, navigate to your remote `feature` branch, and press the
+`Compare & pull request` button.
+
+![submit_pr_create_pr_1.png](./imgs/submit_pr_create_pr_1.png)
+
+The short and long description fields of the pull request are auto-filled using
+the short and long description of your commit.
+
+![submit_pr_create_pr_2.png](./imgs/submit_pr_create_pr_2.png)
+
+A `template` is also provided in the long description field.
+
+Use the template to fill in as much information as possible to properly describe
+the purpose of the pull request.
+
+{% include tip.html content="The title of the pull request should start with <b>[WIP]</b> for <b>work in progress</b> when the pull request is not complete yet." %}
```

Comment:
@dhmlau mentioned GitHub now supports [draft PRs ](https://github.blog/2019-02-14-introducing-draft-pull-requests/) that we can use instead of [WIP].

Reply:
@dmlau , I read about them just now, and there isn't any indication they can be distinguished from normal PRs in the UI. So having [WIP] would still be useful in the title.